docs/aws/audit/snsmonitoring/rules/is_sns_topic_encrypted.mdx

```diff
@@ -23,6 +23,197 @@ HIPAA, GDPR, NIST, HITRUST
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Simple Notification Service (SNS) dashboard by selecting "Services" from the top menu, then typing "SNS" in the search box and selecting "Simple Notification Service".
+3. In the SNS dashboard, select "Topics" from the left-hand navigation pane. This will display a list of all the SNS topics in your AWS account.
+4. For each topic, check the "Encryption" column. If the column is blank or shows "Disabled", then the SNS topic is not encrypted. If it shows "Enabled", then the topic is encrypted.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the SNS topics in your AWS account. You can do this by using the following AWS CLI command:
+
+```bash
+aws sns list-topics
+```
+
+This command will return a list of all SNS topics in your AWS account.
+
+2. Once you have the list of SNS topics, you can check the encryption status of each topic. You can do this by using the following AWS CLI command:
+
+```bash
+aws sns get-topic-attributes --topic-arn <TopicARN>
+```
+
+Replace `<TopicARN>` with the ARN of the SNS topic you want to check. This command will return the attributes of the specified SNS topic.
+
+3. In the output of the above command, look for the `KmsMasterKeyId` attribute. If this attribute is present and has a value, it means that the SNS topic is encrypted.
+
+4. Repeat steps 2 and 3 for each SNS topic in your AWS account. If any SNS topic does not have the `KmsMasterKeyId` attribute or if this attribute does not have a value, it means that the SNS topic is not encrypted.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so. You can install it using pip:
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Use the SNS client from the session to get all the SNS topics:
+```python
+sns = session.client('sns')
+response = sns.list_topics()
+```
+
+4. Iterate over the topics and check if they are encrypted. If the 'KmsMasterKeyId' key is not present in the topic attributes, the topic is not encrypted:
+```python
+for topic in response['Topics']:
+    topic_arn = topic['TopicArn']
+    topic_attributes = sns.get_topic_attributes(TopicArn=topic_arn)['Attributes']
+    if 'KmsMasterKeyId' not in topic_attributes:
+        print(f'Topic {topic_arn} is not encrypted.')
+```
+This script will print the ARNs of all unencrypted SNS topics.
+</Accordion>
+
+</AccordionGroup>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>